ci-operator/step-registry/azure/provision/service-principal/hypershift/azure-provision-service-principal-hypershift-commands.sh

```diff
@@ -25,34 +25,35 @@ declare -A component_to_client_id
 declare -A component_to_cert_name
 
 for component in $COMPONENTS; do
-    name="${SP_NAME_PREFIX}-${component}"
-    scopes="/subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_HC"
-    if [[ $component == ingress ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    elif [[ $component == cloud-provider ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-    elif [[ $component == cpo ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    elif [[ $component == capz ]]; then
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
-        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
-    fi
+   name="${SP_NAME_PREFIX}-${component}"
+# by default, we should enable HYPERSHIFT_AZURE_ASSIGN_SERVICE_PRINCIPAL_ROLES to assign roles to the service principal automatically
+#    scopes="/subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_HC"
+#    if [[ $component == ingress ]]; then
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
+#    elif [[ $component == cloud-provider ]]; then
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
+#    elif [[ $component == cpo ]]; then
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
+#    elif [[ $component == capz ]]; then
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_NSG"
+#        scopes+=" /subscriptions/$AZURE_AUTH_SUBSCRIPTION_ID/resourceGroups/$RG_VNET"
+#    fi
 
-    client_id="$(eval "az ad sp create-for-rbac --name $name --role Contributor --scopes $scopes --create-cert --cert $name --keyvault $KV_NAME --output json --only-show-errors" | jq -r '.appId')"
+    client_id="$(eval "az ad sp create-for-rbac --name $name --create-cert --cert $name --keyvault $KV_NAME --output json --only-show-errors" | jq -r '.appId')"
     echo "$client_id" >> "${SHARED_DIR}/azure_sp_id"
 
     component_to_client_id+=(["$component"]="$client_id")
     component_to_cert_name+=(["$component"]="$name")
 done
 
 # TODO: Remove this once the we used the automated role assignment by "--assign-service-principal-role"
```

Comment:
Should this whole block just be removed now?

Reply:
Here it prepares one managed identity file `hypershift_azure_mi_file.json` used in the azure provision chain when HYPERSHIFT_AZURE_CP_MI is true and it is also needed by HYPERSHIFT_AZURE_ASSIGN_SERVICE_PRINCIPAL_ROLES https://github.com/openshift/release/blob/ba456e469fb8b920ad5c4a9785c85cb2892c7c8c/ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml#L171
https://github.com/openshift/hypershift/blob/92bbcedfa4951be1c4edbba6465342a37fb6fda7/cmd/infra/azure/create.go#L223

And in the hypershift validation, this HYPERSHIFT_AZURE_CP_MI is needed when TechPreviewEnabled is enabled. So I have to keep that logic here. 
https://github.com/openshift/hypershift/blob/001b0daa3ceecb87ab0a3388b4cff40338b7d633/cmd/cluster/azure/create.go#L141

Reply:
@bryan-cox One question: If we don't prepare the service principal in advance, is there another step or a specific area within Hypershift that handles the creation of service principals? Thank you!

Reply:
No, we removed the creation of SPs on purpose as it contributes to Azure quota issues. The direction from Red Hat and Microsoft were to reuse SPs as much as we can.

Reply:
OK, I will prepare some permanent SPs manually and store them into the vault to use them in the step. thanks